bson/raw_bson.py

```diff
@@ -111,7 +126,7 @@ class from the standard library so it can be used like a read-only
         # it refers to this class RawBSONDocument.
         if codec_options is None:
             codec_options = DEFAULT_RAW_BSON_OPTIONS
-        elif codec_options.document_class is not RawBSONDocument:
+        elif codec_options.document_class not in [RawBSONDocument, LazyRawBSONDocument]:
```

Comment:
`not issubclass(codec_options.document_class, RawBSONDocument)` would be more future proof.

Reply:
Done